src/rules/maxLineLengthRule.ts

```diff
@@ -96,32 +108,98 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     private getRuleOptions(): MaxLineLengthRuleOptions {
-        const argument = this.ruleArguments[0];
-        let options: MaxLineLengthRuleOptions = { limit: 0 };
+        const argument = this.ruleArguments[0] as MaxLineLengthRuleOptions | number;
+
         if (typeof argument === "number") {
-            options.limit = argument;
+            return { [OPTION_LIMIT]: argument };
         } else {
-            options = argument as MaxLineLengthRuleOptions;
-            const ignorePattern = (argument as { [key: string]: string })["ignore-pattern"];
-            options.ignorePattern =
-                typeof ignorePattern === "string" ? new RegExp(ignorePattern) : undefined;
+            const {
+                [OPTION_CHECK_REGEX]: checkRegex,
+                [OPTION_CHECK_STRINGS]: checkStrings,
+                [OPTION_IGNORE_PATTERN]: ignorePattern,
+                [OPTION_LIMIT]: limit,
+            } = argument;
```

Comment:
It's rather difficult to parse through `[OPTON_CHECK_REGEX]` destructurings and co. later on in this file. You're using the same verbose rule format in both the _external_ options for the rule (e.g. this line) and the _internal_ options (the type in the `WalkContext`).

Requested change: please switch back the `MaxLineLengthRuleOptions` type to have camelCase keys, so it can be used directly like `options.limit`.